tvb_contrib/tvb/contrib/cosimulation/cosimulator.py

```diff
@@ -312,9 +331,8 @@ def loop_cosim_monitor_output(self, n_steps=None, relative_start_step=0):
                 # check if it's valid input
                 ValueError("Incorrect n_steps = %i; it should be <= %i".format(
                            n_steps, self.good_cosim_update_values_shape[0]))
-            if relative_start_step < 0 or \
```

Comment:
Why did you remove this condition?

Reply:
> Why did you remove this condition?

Allowing negative relative time steps for the loop_cosim_monitor_output method of CoSimunlator, in order to be able to send the TVB state or the initial condition to the other co-simulator from further in the past, , e.g., from 2*synchronization_time in the past.

This is mainly needed for the TVB initial condition:
the [-min_tvb_delay, -min_tvb_delay/2] is sent also to the TVB->spikeNet transformer and at time 0 towards spikeNet,
whereas [-min_tvb_delay/2, 0.0] is sent at time 0 to the TVB->spikeNet transformer

Reply:
Thanks for the comments.

I made the changes, and also made the last additions to the CoSimulator (relative_output_time_steps) and corrected the specific test.

All tests PASS now.